### PR TITLE
Update Hyku to bring in snippets fixes

### PR DESCRIPTION
**Pull request by laritakr**
# Story

Refs

- https://github.com/scientist-softserv/adventist_knapsack/issues/769

# Expected Behavior Before Changes

Snippets didn't work correctly

# Expected Behavior After Changes

- [ ] Search on catalog page performs full text search and shows highlighted snippets.
- [ ] Title and thumbnail urls carry the search terms through to the show pages and perform UV search automatically
- [ ] Search with no highlighting opens show page normally.

# Screenshots / Video

<details>
<summary></summary>

### Search on catalog page picks up terms in both full text and other metadata 
Screenshot 2024-10-25 at 5 30 51 PM
### clicking on work automatically searches the UV
Screenshot 2024-10-25 at 5 31 17 PM

</details>

# Notes

### CODE REVIEW

File: app/controllers/catalog_controller_decorator.rb [hunk 1]
```diff
@@ -62,7 +62,13 @@
     { prop => CatalogController.send(:index_options, prop, DogBiscuits.config.property_mappings[prop]) }
   end
   CatalogController.send(:add_index_field, config, index_props)
-  config.add_index_field 'all_text_tsimv', label: "Item contents", highlight: true, helper_method: :render_ocr_snippets, if: :query_present?
+  # this is identical to what is in Hyku but the dog biscuits code seems to mean it doesn't carry through
```
Comment (ShanaLMoore) [inline app/controllers/catalog_controller_decorator.rb:65]: instead of copying this over you can subtract it some place I saw. Let me find it ...

Comment (ShanaLMoore) [reply app/controllers/catalog_controller_decorator.rb:65]: I think you can add it [here](https://github.com/scientist-softserv/adventist_knapsack/blob/3560ec048061c0610afbec232fde5ec2cede970a/app/controllers/catalog_controller_decorator.rb#L48). 

`next if key == 'all_text_tsimv' and it'll fall back to hyku`

File: app/helpers/hyku_knapsack/application_helper.rb [hunk 2]
```diff
@@ -15,25 +15,6 @@ def video_embed_viewer_display(work_presenter, locals = {})
...
-    # This is in place to coerce the :q string to :query for passing the :q value to the query value
-    # of a IIIF Print manifest.
-    #
-    # @param doc [SolrDocument]
-    # @param request [ActionDispatch::Request]
-    def generate_work_url(doc, request)
-      url = super
-      return url if request.params[:q].blank?
-
-      key = doc.any_highlighting? ? 'parent_query' : 'query'
```
Comment (ShanaLMoore) [inline app/helpers/hyku_knapsack/application_helper.rb:29]: Do we know what the diff of sending parent_query and query is? I don't think Hyku sends query

Comment (laritakr) [reply app/helpers/hyku_knapsack/application_helper.rb:29]: That's why I deleted this... we refactored hyku and this was no longer applicable.

File: app/controllers/catalog_controller_decorator.rb (hunk 1 above)
Comment (laritakr) [reply app/controllers/catalog_controller_decorator.rb:65]: Good suggestion, thanks! I'll make that change.

Comment (laritakr) [reply app/controllers/catalog_controller_decorator.rb:65]: Actually it is already there (line 49), and it still wasn't working, so I think this is needed.

**Review by ShanaLMoore — APPROVED** [review]

**Pull request merged**